### PR TITLE
feat: accept multiple project ids by providing a comma separate list of ids

### PR DESCRIPTION
**Pull request by lo1tuma**

### CODE REVIEW

**Review by screendriver — CHANGES_REQUESTED** [review]

File: src/config.ts [hunk 1]
```diff
@@ -1,44 +1,68 @@
...
-    return configSchema.parse(loaded?.config);
+    return partialConfigInputSchema.parse(loaded?.config);
   } catch {
     return undefined;
   }
 }
 
-export type CliArguments = Config;
+function convertToNumber(value: string): number {
+  return parseInt(value.trim(), 10);
```
[the remark this answers is not in the document]
Comment (lo1tuma) [reply src/config.ts]: Cool

File: src/config.ts [hunk 2]
```diff
@@ -1,44 +1,68 @@
...
+}
+
+function commaSeparatedStringToNumberArray(commaSeparatedString: string): number[] {
```
[the remark this answers is not in the document]
Comment (lo1tuma) [reply src/config.ts]: done

File: src/App.tsx [hunk 3]
```diff
@@ -2,13 +2,14 @@ import React, { FunctionComponent, useEffect, useState } from 'react';
...
 import throttle from 'p-throttle';
+import pMap from 'p-map';
 import type { ListPipelinesFunction, DeletePipelineFunction, FilterPipelinesByDateFunction } from './gitlab';
 import { Error, ErrorProps } from './Error';
 import { Pipeline } from './network';
 
 export interface AppProps {
   gitlabUrl: string;
-  projectId: number;
+  projectIds: number[];
```
[the remark this answers is not in the document]
Comment (lo1tuma) [reply src/App.tsx]: done

File: src/config.ts [hunk 4]
```diff
@@ -1,44 +1,68 @@
...
   gitlabUrl: z.string().url(),
-  projectId: z.number(),
   accessToken: z.string(),
   days: z.number(),
   trace: z.boolean(),
 });
 
-const configSchema = requiredArguments.partial();
+const configSchema = baseConfigSchema.extend({
+  projectIds: z.array(z.number()).nonempty(),
```
[the remark this answers is not in the document]
Comment (lo1tuma) [reply src/config.ts]: Done

**Review by screendriver — APPROVED** [review]
GNTM 👠

**Review by screendriver — APPROVED** [review]

**Pull request merged**